lib/core/ingestion/es_sink.rb

```diff
@@ -11,49 +11,108 @@
 require 'utility/es_client'
 require 'utility/logger'
 require 'elasticsearch/api'
-
+#
+# This class is responsible for sending the data to the data storage.
+# While we don't actually allow to output our data anywhere except
+# Elasticsearch, we still want to be able to do so sometime in future.
+#
+# This class should stay simple and any change to the class should be careful
+# with the thought of introducing other sinks in future.
 module Core
   module Ingestion
     class EsSink
-      def initialize(index_name, request_pipeline, bulk_queue = Utility::BulkQueue.new)
+      def initialize(index_name, request_pipeline, bulk_queue = Utility::BulkQueue.new, max_allowed_document_size = 5 * 1024 * 1024)
         @client = Utility::EsClient.new(App::Config[:elasticsearch])
         @index_name = index_name
         @request_pipeline = request_pipeline
         @operation_queue = bulk_queue
+
+        @max_allowed_document_size = max_allowed_document_size
+
+        @queued = {
+          :indexed_document_count => 0,
+          :deleted_document_count => 0,
+          :indexed_document_volume => 0
+        }
+
+        @completed = {
+          :indexed_document_count => 0,
+          :deleted_document_count => 0,
+          :indexed_document_volume => 0
+        }
       end
 
-      def ingest(id, serialized_document)
-        index_op = serialize({ 'index' => { '_index' => index_name, '_id' => id } })
+      def ingest(document)
+        unless document&.any?
```

Comment:
```suggestion
        unless document&.present?
```

Nitpick: `.any?` feels like something to use on a collection, but "document" is singular. Technically it works for a hash, but reads kinda funny IMO

Reply:
`.present` is a Rails method, I had a comment from someone to try to not use Rails method here.

Another option would be to use `.empty?` method but it'll require flipping the condition that'll look uglier probably:

```
unless document || document.empty? == false
```

or 

```
if document.nil? || document.empty? # this looks pretty okay, gotta use it!
```

Reply:
Actually, it's from [activesupport](https://rubygems.org/gems/activesupport), which, is extracted from Rails.

We have `activesupport` as a gem dependency, if you want to use it. But I don't have preference between `any?` and `present?`

Reply:
@vidok I think it may have been you who discourages activesupport usage. Why is that? I find these types of helpers really valuable, and since the majority of ruby users are rails users, the StackOverflow snippets we encounter often assume active_support is available. Why avoid it?